pex/pip.py

```diff
@@ -588,18 +611,179 @@ def create(
                 isolated_pip_builder.info.venv = True
                 for dist_location in third_party.expose(["pip", "setuptools", "wheel"]):
                     isolated_pip_builder.add_dist_location(dist=dist_location)
+                env_var_names = dict(
+                    patched_markers_env_var_name=cls._PATCHED_MARKERS_FILE_ENV_VAR_NAME,
+                    skip_markers_env_var_name=cls._SKIP_MARKERS_ENV_VAR_NAME,
+                    python_versions_env_var_name=cls._PYTHON_VERSIONS_FILE_ENV_VAR_NAME,
+                )
                 with named_temporary_file(prefix="", suffix=".py", mode="w") as fp:
                     fp.write(
                         dedent(
                             """\
```

Comment:
This dynamic patching only has 4 items, but its pretty large. I'd like to follow up and break the patches out to a (re)source file: #1581

Reply:
Its not clear to me at all his work deals with this universal stuff. I don't think it does. That's why I pinged. I think there are really just 2 ways to lessen maintenance burden:
1. Pip comes out with lockfile support that suits Pex needs and we switch to that.
2. Pex moves away from Pip and builds around resolvelib directly.

Given my experience with Pip maintainers (and even that aside and putting myself in their shoes) and I think both are about as hard / unlikely to happen so I'd lean towards #2 without having given it alot of thought.